packages/grpc-js-core/src/call-credentials-filter.ts

```diff
@@ -7,13 +7,27 @@ import {BaseFilter, Filter, FilterFactory} from './filter';
 import {Metadata} from './metadata';
 
 export class CallCredentialsFilter extends BaseFilter implements Filter {
-  constructor(private readonly credentials: CallCredentials) {
+  private serviceUrl: string;
+  constructor(
+      private readonly channel: Http2Channel,
+      private readonly host: string,
+      private readonly path: string) {
     super();
+    let splitPath: string[] = path.split('/');
+    let serviceName: string = '';
+    /* The standard path format is "/{serviceName}/{methodName}", so if we split
+     * by '/', the first item should be empty and the second should be the
+     * service name */
+    if (splitPath.length >= 2) {
```

Comment:
Is `serviceName` meant to be empty if this is not the case?

Reply:
yes